src/jit/codegenarm64.cpp

```diff
@@ -2943,13 +2950,25 @@ CodeGen::genCodeForTreeNode(GenTreePtr treeNode)
             assert(cmp->OperIsCompare());
             assert(compiler->compCurBB->bbJumpKind == BBJ_COND);
 
-            // Get the "jmpKind" using the gtOper kind
-            // Note that whether it is an unsigned cmp is governed by the GTF_UNSIGNED flags
+            // Get the "kind" and type of the comparison.  Note that whether it is an unsigned cmp
+            // is governed by a flag NOT by the inherent type of the node
+            emitJumpKind jumpKind[2];
+            bool branchToTrueLabel[2];
+            genJumpKindsForTree(cmp, jumpKind, branchToTrueLabel);
```

Comment:
It would be nice to have an assert to the effect of `jumpKind[0] != EJ_NONE || jumpKind[1] != EJ_NONE` here.